src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,25 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_properties_from_model_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    output_document = {"properties": output_model.copy()}
+    input_document = {"properties": input_model.copy()}
+    for path in paths:
+        try:
+            _prop, _resolved_path, _parent = traverse(input_document, path)
+        except LookupError:
+            _prop, resolved_path, parent = traverse(output_document, path)
```

Comment:
what if the path does not exist in the output_document?  It could happen if Create doesn't return the createOnlyProperties. But it will fail in an ugly way.   We should catch it and give proper error message.

Reply:
makes sense. I will add that check